substrate/client/transaction-pool/src/graph/base_pool.rs

```diff
@@ -453,27 +453,29 @@ impl<Hash: hash::Hash + Member + Serialize, Ex: std::fmt::Debug> BasePool<Hash,
 
 		while ready.is_exceeded(self.ready.len(), self.ready.bytes()) {
 			// find the worst transaction
-			let worst = self.ready.fold::<TransactionRef<Hash, Ex>, _>(|worst, current| {
-				let transaction = &current.transaction;
-				worst
-					.map(|worst| {
-						// Here we don't use `TransactionRef`'s ordering implementation because
-						// while it prefers priority like need here, it also prefers older
-						// transactions for inclusion purposes and limit enforcement needs to prefer
-						// newer transactions instead and drop the older ones.
-						match worst.transaction.priority.cmp(&transaction.transaction.priority) {
-							Ordering::Less => worst,
-							Ordering::Equal =>
-								if worst.insertion_id > transaction.insertion_id {
-									transaction.clone()
-								} else {
-									worst
-								},
-							Ordering::Greater => transaction.clone(),
-						}
-					})
-					.or_else(|| Some(transaction.clone()))
-			});
+			let worst =
+				self.ready.fold::<Option<TransactionRef<Hash, Ex>>, _>(None, |worst, current| {
+					let transaction = &current.transaction;
+					worst
+						.map(|worst| {
+							// Here we don't use `TransactionRef`'s ordering implementation because
+							// while it prefers priority like need here, it also prefers older
+							// transactions for inclusion purposes and limit enforcement needs to
+							// prefer newer transactions instead and drop the older ones.
+							match worst.transaction.priority.cmp(&transaction.transaction.priority)
+							{
+								Ordering::Less => worst,
+								Ordering::Equal =>
+									if worst.insertion_id > transaction.insertion_id {
```

Comment:
I see this `insertion_id` of type u64 that comes with the ready txs set of a pool, which gets incremented with each new ready tx imported. It will take a while until this can overflow and flip the logic for a bit (until all the upper end u64 TXs of the ready set are removed), but are there other reasons that makes this harder than having u64::MAX imported TXs that go through the ready set?

Reply:
Not sure I understood a question. Would you please rephrase it?

Reply:
The question is are we concerned with `insertion_id` being wrapped around (U64::MAX -> 0)? In this case an insertion_id = 0 can be considered more recent than a previous insertion_id e.g U64::MAX, so the logic would not stand for this situation.

Reply:
Well, I'd say no. 2^64 is quite a huge number.

The other thing is that this code existed for a while before I started to stick my nose into txpool. So maybe - in future we should update this code to have here a value that is exactly the same across different views (timestamp from `TimedTransactionSource` could be a good candidate).